sdk/src/introspection.ts

```diff
@@ -16,3 +20,35 @@ export function getInterfaceId(iface: Interface): string {
   }
   return interfaceId.toHexString();
 }
+
+/**
+ * Gets the protocol version of a contract, if the contract does not have a
+ * protocolVersion function, it will return [1, 0, 0]
+ *
+ * @export
+ * @param {string} rpc
+ * @param {string} contractAddress
+ * @return {*}  {Promise<[number, number, number]>}
+ */
+export async function getProtocolVersion(
+  rpc: string,
+  contractAddress: string
+): Promise<[number, number, number]> {
+  if (!isAddress(contractAddress)) {
+    throw new InvalidAddressError(contractAddress);
+  }
+  const provider = new JsonRpcProvider(rpc);
+  const iface = new Interface([
+    'function protocolVersion() public pure returns (uint8[3] memory)',
+  ]);
+  const contract = new Contract(contractAddress, iface, provider);
+  let version: [number, number, number];
+  try {
+    version = await contract.protocolVersion();
+  } catch (e) {
+    // ethers5 throws a call exception error which could mean a lot of things
+    // so this is not accurate
```

Comment:
what does this message mean? does it mean it should be managed differently?
if the error could be due to multiple reasons why only set a predefined version?

Reply:
So the problem is that the v1 contracts dont have a way of knowing they are v1.0.0,  so the error thrown by ethers is way to generic to be able to say if its throwing becuase it does nof find the function and indeed is a v1 contract or any other reason, so in the past we agreed to return 1.0.0 if it throws, nobody liked the solution

Reply:
Maybe we need a better description here

Reply:
ok, so versions were not supported on the first OSx version and now it is. 
and since `protocolVersion` is a view and doesn't return/revert/assert this code assumes the only possible error is that the function doesn’t exist like in v1. Is this right? 
In that case, place a more explicit comment